src/mavoscript.js

```diff
@@ -321,7 +321,62 @@ var _ = Mavo.Script = {
 				return ret;
 			},
 			precedence: 3
-		}
+		},
+		"groupby": {
+			symbol: "by",
+			code: (array, key) => {
+				if (Array.isArray(array) && Array.isArray(key))  { 
+					var propName = key[Mavo.as] === true ? key[Mavo.name] : $.value(key[0], Mavo.toNode, "property");
+					var temp = new Mavo.BucketMap({arrays: true});
+					var ret = [];
+					ret[Mavo.groupedBy] = true;
+
+					for (var i = 0; i < array.length; i++) {
+						const k = i < key.length ? Mavo.value(key[i]) : null;
+						temp.set(k, Mavo.value(array[i]));
+					}	
+
+					for (var [value, items] of temp) {
+						var obj = {"$value": value};
+
+						if (propName !== undefined) {
+							obj[propName] = value;
+						}
+						
+						obj["$items"] = items;
+						ret.push(obj);
+					}
+
+					return ret;
+				}
+				else { // invalid arguments, convert to arrays
+					// return (Mavo.toArray(array), Mavo.toArray(key));
```

Comment:
What happened here?
I'd just do this in the beginning of the function: `array = Array.isArray(array)? array : Mavo.toArray(array);` and same for key and then move on.

Reply:
Oops - I meant to fix this and guess I forgot. Fixing it now.